src/state/machines/turn.ts

```diff
@@ -324,19 +349,25 @@ export const TurnMachine = setup({
     },
     buying: {
       on: {
-        "user.click.token": {
-          target: "#turn.ability",
-          actions: assign({
-            turn: ({ context: { turn }, event: { token } }) => ({
-              ...turn,
-              currentAbility: { piece: token, name: token.name },
+        "user.click.token": [
+          {
+            target: "#turn.ability",
+            actions: assign({
+              turn: ({ context: { turn }, event: { token } }) => ({
+                ...turn,
+                currentAbility: { piece: token, name: token.name },
+              }),
             }),
-          }),
-          guard: {
-            type: "abilityAvailable",
-            params: ({ event: { token } }) => token,
+            guard: {
+              type: "abilityAvailable",
+              params: ({ event: { token } }) => token,
+            },
           },
-        },
+          {
+            actions: { type: "refreshAbility", params: ({ event: { token } }) => token.uid },
+            guard: "canRefreshAbility",
+          },
+        ],
```

Comment:
hm.. don't we want this to happen automatically?

Reply:
Technically no because the user should decide which one to refresh